app/src/main/java/com/google/android/sambadocumentsprovider/nativefacade/SambaFacadeClient.java

```diff
@@ -81,7 +82,7 @@ public void reset() {
   public StructStat stat(String uri) throws IOException {
     try (final MessageValues<StructStat> messageValues = MessageValues.obtain()) {
       final Message msg = obtainMessage(STAT, messageValues, uri);
-      enqueue(msg);
+      processMessage(msg);
```

Comment:
I'm pretty sure this isn't right. In Android O stat() will be executed in-place, but it's still called from other places to obtain file stats which is not in the correct thread.

Maybe it's because you still have misunderstanding on the implementation of fuse & SAF.